PatternPal/PatternPal.Extension/ViewModels/StepByStepListViewModel.cs

```diff
@@ -29,17 +30,41 @@ public class StepByStepListViewModel : ViewModel
             NavigateHomeCommand = new NavigateCommand< HomeViewModel >(
                 navigationStore,
                 () => new HomeViewModel(navigationStore));
+            
+            // Next button 
             NavigateStepByStepInstructionsCommand =
                 new NavigateCommand< StepByStepInstructionsViewModel >(
                     navigationStore,
                     () => new StepByStepInstructionsViewModel(
                         navigationStore,
-                        SelectedInstructionSet));
+                        SelectedInstructionSet,
+                        false));
+            
+            // Continue button
+            NavigateContinueStepByStepInstructionsCommand =
+                new NavigateCommand<StepByStepInstructionsViewModel>(
+                    navigationStore,
+                    () => new StepByStepInstructionsViewModel(
+                        navigationStore,
+                        SelectedInstructionSet,
+                        true));
+
+            GetInstructionSetsResponse response = 
+                GrpcHelper.StepByStepClient.GetInstructionSets(new GetInstructionSetsRequest());
+
+            List<Recognizer> temFilteredRecognizers = response.Recognizers.Where(r => r is Recognizer.Singleton).ToList();
```

Comment:
Why are you doing this filtering on the frontend? Even if it's temporary, it would still be better to do it in the background service.

Reply:
Filtering has been moved to the GetInstructionSet in the StepByStepSerivce.